release/ray_release/util.py

```diff
@@ -42,7 +42,11 @@ def dict_hash(dt: Dict[Any, Any]) -> str:
 
 
 def url_exists(url: str) -> bool:
-    return requests.head(url, allow_redirects=True).status_code == 200
+    try:
+        return requests.head(url, allow_redirects=True).status_code == 200
+    except requests.exceptions.RequestException:
+        logger.exception(f"Failed to check url exists: {url}")
+        return False
```

Comment:
Can we log the exception here, too?

```suggestion
    except requests.exceptions.RequestException as e:
        logger.exception(f"Failed to check if URL exists ({url}): {e}")
        return False
```

Reply:
`logger.exception` should be enough